File: SupportSDK/Examples/Objective-C/SupportSDK-ObjC/ViewController.m
//
//  ViewController.m
//  SupportSDK-ObjC
//
//  Created by Larry Borsato on 2019-09-17.
//  Copyright © 2019 Boomtown. All rights reserved.
//

#import "ViewController.h"

@interface ViewController () <SupportButtonDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"Support SDK Demo";
    
    CGFloat buttonSize = 75;
    CGRect buttonFrame = CGRectMake((self.view.frame.size.width-buttonSize)/2,
                                    (self.view.frame.size.height-buttonSize)/2,
                                    buttonSize,
                                    buttonSize);
    SupportButton *supportButton = [[SupportButton alloc] initWithFrame:buttonFrame];
    supportButton.delegate = self;
    if ( ![supportButton loadConfigurationFile:@"CONFIGURATION FILE NAME" customerInfo:nil] ) {
        NSLog(@"Unable to load configuration file");
    }
	[self.view addSubview:supportButton];
    
    [supportButton advertiseServiceWithPublicData:@{@"test":@"data"}
                                           privateData:@{@"test":@"private data"}];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
}


#pragma mark - Support button delegates

- (void) supportButton:(nonnull SupportButton *)supportButton didAdvertiseService:(NSNetService *)netService
{
    NSLog(@"mDNS service successfully advertised.");
}


- (void) supportButton:(nonnull SupportButton *)supportButton didFailToAdvertiseService:(NSDictionary<NSString *,NSNumber *> *)errorDict
{
    NSLog(@"mDNS service failed to advertise.");
}


- (void) supportButton:(nonnull SupportButton *)supportButton  displaySupportMenu:(nonnull UIAlertController *)alertController
{
    if ( self.view.traitCollection.horizontalSizeClass != UIUserInterfaceSizeClassCompact )
    {
        [self presentViewController:alertController animated:YES completion:nil];
		UIPopoverPresentationController *popPresenter = [alertController
														 popoverPresentationController];
		popPresenter.sourceView = supportButton;
		popPresenter.sourceRect = supportButton.bounds;
	}
	else
	{
		[self.navigationController presentViewController:alertController animated:YES completion:nil];
	}
}


- (void) supportButton:(nonnull SupportButton *)supportButton displayIssueViewController:(UIViewController *)viewController
{
	[self.navigationController pushViewController:viewController animated:YES];
}


- (void) supportButton:(nonnull SupportButton *)supportButton didFailWithError:(nonnull NSError *)error
{
	NSLog(@"%@: %@", error.localizedDescription, error.localizedFailureReason);
}


@end

File: SupportSDK/Examples/Swift/SupportSDK-Swift/ViewController.swift
//
//  ViewController.swift
//  SupportSDK-Swift
//
//  Created by Larry Borsato on 2019-09-17.
//  Copyright © 2019 Boomtown. All rights reserved.
//

import UIKit
import SupportSDK

class ViewController: UIViewController, SupportButtonDelegate {
   
    lazy var supportButton: SupportButton = {
        [unowned self] in
        
        let buttonSize = 75
        let buttonFrame = CGRect.init(x: 0,
                                      y: 0,
                                      width: buttonSize,
                                      height: buttonSize)
        let button = SupportButton(frame: buttonFrame)
        button.delegate = self;

        if button.loadConfigurationFile("CONFIGURATION FILE NAME", customerId: nil) == false  {
            NSLog("Unable to load configuration file")
        }
        return button
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        title = "Support SDK Demo"
        
        view.addSubview(supportButton)
        var frame = supportButton.frame
        frame.origin.x = (self.view.frame.size.width-frame.size.width)/2
        frame.origin.y = (self.view.frame.size.height-frame.size.width)/2
        supportButton.frame = frame
        
        supportButton.advertiseService(withPublicData: ["test": "data"], privateData: ["test": "private data"])
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
    }


    func supportButton(_ supportButton: SupportButton, displayIssueViewController viewController: UIViewController) {
        show(viewController, sender: supportButton)
    }
    
    func supportButton(_ supportButton: SupportButton, displaySupportMenu alertController: UIAlertController) {
        present(alertController, animated: true, completion: nil)
        if let popoverController = alertController.popoverPresentationController {
            popoverController.sourceView = supportButton
            popoverController.sourceRect = supportButton.bounds
        }
    }
    
    func supportButton(_ supportButton: SupportButton, didFailWithError error: Error) {
        NSLog("%@", error.localizedDescription)
    }
    
    func supportButton(_ supportButton: SupportButton, didAdvertiseService netService: NetService) {
            NSLog("mDNS service successfully advertised.");
    }
    
    func support(_ supportButton: SupportButton, didFailToAdvertiseService errorDict: [String : NSNumber]) {
        NSLog("mDNS service failed to advertise.");
    }

}

